src/logs/coralogix.rs

```diff
@@ -269,19 +317,46 @@ fn convert_to_log_entry(
 ) -> LogSinglesEntry<Value> {
     let now = OffsetDateTime::now_utc();
     
-    let application_name = mctx
-        .evaluate(configured_app_name.to_string())
-        .unwrap_or_else(|e| {
-            tracing::warn!("application name dynamic parsing failed, using: {}", e);
-            configured_app_name.to_owned()
-        });
+    let application_name = if configured_app_name.starts_with("{{ json.") && configured_app_name.ends_with(" }}") {
+        // Extract the JSON path from between "{{ json." and " }}"
+        let key_path = configured_app_name
+            .trim_start_matches("{{ json.")
+            .trim_end_matches(" }}")
+            .trim();
+            
+        extract_json_value(&log, key_path)
+            .unwrap_or_else(|| {
+                tracing::warn!("application name JSON parsing failed, using configured value");
+                configured_app_name.to_owned()
+            })
+    } else {
+        mctx.evaluate(configured_app_name.to_string())
+            .unwrap_or_else(|e| {
+                tracing::warn!("application name dynamic parsing failed, using: {}", e);
+                configured_app_name.to_owned()
+            })
+    };
 
+    let subsystem_name = if configured_sub_name.starts_with("{{ json.") && configured_sub_name.ends_with(" }}") {
```

Comment:
We should probably use a regex here for the match, rather than starts_with since the user might use something like {{json.****}} or {{    json.xxxx   }. By using a regex we can match if they uses spaces or not.